styles/stripes.h

```diff
@@ -47,6 +55,24 @@ class StripesHelper<A, B...> {
     }
     b_.get(led, p - 341, ret);
   }
+  void hardGet(int led, int p, SimpleColor* ret) {  // Add hardGet method
+    int segment_size = 341;
+    if (p < segment_size) {
+      OverDriveColor tmp = a_.getColor(led);
+      if (p > 0 && p < 512) { // One-pixel transition logic
```

Comment:
This seems wonky to me, pixels are not 512 units, and there shouldn't be any need for sin table stuff here.

I recommend trying something like this:

void hardGet(Range pixel, int led, int p, uint32_t mod, SimpleColor *ret)
  int segment_size = 341;
  uint32_t weight = pixel.intersect_with_stripe(Range(p, p+segment_size), mod);
  if (weight) {
    OverDriveColor tmp = a_.getColor(led);
    if (weight == pixel.size()) {
        ret->c = tmp.c;
        return;
    } else {
      weight = weight * 16384 / pixel.size();
      ret->c.r = clampi32(ret->c.r + ((tmp.c.r * weight) >> 14), 0, 65535);
      ret->c.g = clampi32(ret->c.g + ((tmp.c.g * weight) >> 14), 0, 65535);
      ret->c.b = clampi32(ret->c.b + ((tmp.c.b * weight) >> 14), 0, 65535);
  }
  b_.hardGet(pixel, led, p + segment_size, mod, ret);
}

And below, when you need to call hardGet, call it like:

  this->colors_.hardGet(Range(0, this->mult_ >> 10), led, p, this->colors_.size() * 341, &ret);

Reply:
what would `intersect_with_stripe` look like?

Reply:
Looks like I forgot to "git push", now you can find it here:

https://github.com/profezzorn/ProffieOS/blob/be38b7a038827f4fef73dffc6d5828eaa89105aa/common/range.h#L37

Reply:
ok how about now?
Tested, looks right on blade.